plugins/woocommerce/templates/order/order-details.php

```diff
@@ -27,9 +27,11 @@
 
 $order_items           = $order->get_items( apply_filters( 'woocommerce_purchase_order_item_types', 'line_item' ) );
 $show_purchase_note    = $order->has_status( apply_filters( 'woocommerce_purchase_note_order_statuses', array( 'completed', 'processing' ) ) );
-$show_customer_details = is_user_logged_in() && $order->get_user_id() === get_current_user_id();
 $downloads             = $order->get_downloadable_items();
 
+// We make sure the order belongs to the user. This will also be true if the user is a guest, and the order belongs to a guest (userID === 0).
+$show_customer_details = $order->get_user_id() === get_current_user_id();
+
```

Comment:
If this is important and needs to be here? can it be done via a hook? otherwise a version bump is needed.

Reply:
Nice catch 👍 , bumped template version in https://github.com/woocommerce/woocommerce/pull/47477/commits/7f95e1ebeee12d68ef22b01ad2b23f786ea7ef29